# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/ipc/VRLayerChild.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 gfx/vr/ipc/VRLayerChild.h
--- a/gfx/vr/ipc/VRLayerChild.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/gfx/vr/ipc/VRLayerChild.h	Mon Jun 15 18:25:55 2020 +0000
@@ -50,7 +50,7 @@
   virtual void ActorDestroy(ActorDestroyReason aWhy) override;
 
   RefPtr<dom::HTMLCanvasElement> mCanvasElement;
-  bool mIPCOpen;
+  bool mIPCOpen = false;
 
   // AddIPDLReference and ReleaseIPDLReference are only to be called by
   // CreateIPDLActor and DestroyIPDLActor, respectively. We intentionally make
@@ -64,10 +64,10 @@
   gfx::Rect mRightEyeRect;
   RefPtr<WebGLFramebufferJS> mFramebuffer;
 
-  RefPtr<layers::SharedSurfaceTextureClient> mThisFrameTexture;
-  RefPtr<layers::SharedSurfaceTextureClient> mLastFrameTexture;
+  Maybe<layers::SurfaceDescriptor> mThisFrameTextureDesc;
+  Maybe<layers::SurfaceDescriptor> mLastFrameTextureDesc;
 
-  uint64_t mLastSubmittedFrameId;
+  uint64_t mLastSubmittedFrameId = 0;
 };
 
 }  // namespace gfx